src/wirecloud/fiware/proxy.py

```diff
@@ -84,50 +131,21 @@ def process_request(self, request):
 
-        if 'fiware-oauth-get-parameter' in request['headers'] or 'x-fi-ware-oauth-get-parameter' in request['headers']:
-            if 'fiware-oauth-get-parameter' in request['headers']:
-                parameter_name = request['headers']['fiware-oauth-get-parameter']
-                del request['headers']['fiware-oauth-get-parameter']
-            else:  # 'x-fi-ware-oauth-get-parameter'
-                parameter_name = request['headers']['x-fi-ware-oauth-get-parameter']
-                del request['headers']['x-fi-ware-oauth-get-parameter']
-
-            url = request['url']
-            if '?' in url:
-                url += '&'
-            else:
-                url += '?'
-
-            url += urlquote_plus(parameter_name) + '=' + urlquote_plus(token)
-            request['url'] = url
+        if 'fiware-oauth-token' in filtered or 'x-fi-ware-oauth-token' in filtered:
+            replace_get_parameter(request, ["fiware-oauth-get-parameter", "x-fi-ware-oauth-get-parameter"], token)
+            replace_header_name(request, ["fiware-oauth-header-name", "x-fi-ware-oauth-header-name"], token)
+            replace_body_pattern(request, ["fiware-oauth-body-pattern", "x-fi-ware-oauth-body-pattern"], token)
 
-        if 'fiware-oauth-header-name' in request['headers'] or 'x-fi-ware-oauth-header-name' in request['headers']:
-            if 'fiware-oauth-header-name' in request['headers']:
-                header_name = request['headers']['fiware-oauth-header-name']
-                del request['headers']['fiware-oauth-header-name']
-            else:  # 'x-fi-ware-oauth-header-name'
-                header_name = request['headers']['x-fi-ware-oauth-header-name']
-                del request['headers']['x-fi-ware-oauth-header-name']
-
-            if header_name == "Authorization":
-                token_pattern = "Bearer {token}"
-            else:
-                token_pattern = "{token}"
-            request['headers'][header_name] = token_pattern.format(token=token)
-
-        if 'fiware-oauth-body-pattern' in request['headers'] or 'x-fi-ware-oauth-token-body-pattern' in request['headers']:
-            if 'fiware-oauth-body-pattern' in request['headers']:
-                pattern = request['headers']['fiware-oauth-body-pattern']
-                del request['headers']['fiware-oauth-body-pattern']
-            else:  # 'x-fi-ware-oauth-token-body-patter'
-                pattern = request['headers']['x-fi-ware-oauth-token-body-pattern']
-                del request['headers']['x-fi-ware-oauth-token-body-pattern']
-
-            new_body = request['data'].read().replace(pattern.encode('utf8'), token.encode('utf8'))
-            request['headers']['content-length'] = "%s" % len(new_body)
-            request['data'] = BytesIO(new_body)
+        if 'fiware-openstack-token' in filtered:
+            # replace_get_parameter(request, ["openstack-oauth-get-parameter"], token)
```

Comment:
Rename `openstack-oauth-get-parameter` into `fiware-openstack-get-parameter`, `openstack-oauth-header-name` into `fiware-openstack-header-name` and `openstack-oauth-body-pattern` into `fiware-openstack-body-pattern`.

And... Why is the `replace_get_parameter` line commented?